### ISSUE TITLE
[Bug] krun - --dry-run doesn't save temporary files

### ISSUE THREAD

**Issue by andreiburdusa** [runtimeverification/k#1850]
K Version
---------

```
$ kompile --version
RV-K version 1.0-SNAPSHOT
Git revision: 4c3c90e
Git branch: add-ref-default.nix
Build date: Tue Mar 30 14:29:27 EEST 2021
```

Description
-----------

Running `krun` with `--dry-run` prints a command that can't be executed because some temporary files are missing.
But those files are present after executing the command with `--save-temps`.

Input Files
-----------

File: `test.k`

```
module TEST
    imports INT
    syntax FOO ::= "foo"
    configuration <k> foo ~> $PGM:Int </k>
    rule <k> foo => #Bottom ... </k>
endmodule
```

File: `2.test`

```
1
```

Reproduction Steps
------------------

```
$ kompile --backend haskell -d . test.k
[Warning] Compiler: Could not find main syntax module with name TEST-SYNTAX in
definition.  Use --syntax-module to specify one. Using TEST as default.
$ krun --search-final --dry-run --haskell-backend-command /home/andrei/kore/.build/kore/bin/kore-exec -d . 2.test
/home/andrei/kore/.build/kore/bin/kore-exec ./test-kompiled/definition.kore --module TEST --pattern .krun-2021-03-11-17-48-49-xUJZdjQeBL/tmp.in.zXImruBIo3 --output .krun-2021-03-11-17-48-49-xUJZdjQeBL/result.kore --searchType FINAL --search .krun-2021-03-11-17-48-49-xUJZdjQeBL/tmp.pattern.PmSzl0lJTD
```
Then execute the resuled command. In this case it's
```
$ /home/andrei/kore/.build/kore/bin/kore-exec ./test-kompiled/definition.kore --module TEST --pattern .krun-2021-03-11-17-48-49-xUJZdjQeBL/tmp.in.zXImruBIo3 --output .krun-2021-03-11-17-48-49-xUJZdjQeBL/result.kore --searchType FINAL --search .krun-2021-03-11-17-48-49-xUJZdjQeBL/tmp.pattern.PmSzl0lJTD
.krun-2021-03-11-17-48-49-xUJZdjQeBL/tmp.pattern.PmSzl0lJTD: copyFile:atomicCopyFileContents:withReplacementFile:copyFileToHandle:openBinaryFile: does not exist (No such file or directory)
```

Expected Behavior
-----------------

The command printed by using `--dry-run` should give the same result as
```
$ krun --search-final  --haskell-backend-command /home/andrei/kore/.build/kore/bin/kore-exec -d . 2.test
```
which is
```
#Bottom
```

**Comment by andreiburdusa** [comment]
I just realised these reproduction steps will work only after https://github.com/kframework/kore/pull/2451 is merged into the backend.

**Comment by radumereuta** [comment]
@andreiburdusa are the reproduction steps the same?

**Comment by andreiburdusa** [comment]
> @andreiburdusa are the reproduction steps the same?

I updated them now by adding the kompilation step

**Comment by ehildenb** [comment]
The new `krun` has the `--save-temps` option too. Maybe that's interacting poorly with `--dry-run`?
